src/Workspaces/Remote/Core/RemoteHostAssetSerialization.cs

```diff
@@ -114,7 +114,6 @@ public static async ValueTask<ImmutableArray<(Checksum, object)>> ReadDataAsync(
                 finally
                 {
                     await localPipe.Writer.CompleteAsync(exception).ConfigureAwait(false);
-                    await pipeReader.CompleteAsync(exception).ConfigureAwait(false);
```

Comment:
>wait pipeReader.CompleteAsyn [](start = 21, length = 28)

Maybe add a comment explaining the policy of who is responsible for closing what.

Reply:
Added comment